nilearn/decomposition/multi_pca.py

```diff
@@ -306,11 +321,8 @@ def fit(self, imgs, y=None, confounds=None):
                                                           subject_pca.shape[0]))
                 data[index * self.n_components:
                      (index + 1) * self.n_components] = subject_pca
-            data, variance, _ = cache(randomized_svd,
-                                self.memory,
-                                func_memory_level=3,
-                                memory_level=self.memory_level)(
-                        data.T, n_components=self.n_components)
+            data, variance, _ = self._cache(randomized_svd, func_memory_level=3)(
```

Comment:
I think this should be put to 2, like in `session_pca`. `func_memory_level > 1` means that intermediary results are cached as well, but I don't really know how subtle we can be on the memory usage.